xaitk_saliency/impls/gen_detector_prop_sal/drise_scoring.py

```diff
@@ -108,9 +108,9 @@ def generate(
                       ref_dets[:, :4]).reshape(n_masks, n_props, n_dets)
 
         # Compute similarity of class probabilities
-        s2 = cdist(perturbed_dets[:, :, 5:].reshape(n_masks * n_props, -1),
-                   ref_dets[:, 5:],
-                   metric=self.proximity_metric).reshape(n_masks, n_props, n_dets)
+        s2 = 1 - cdist(perturbed_dets[:, :, 5:].reshape(n_masks * n_props, -1),
```

Comment:
The `1 - ` component of this may want to be replaced with a uint8 1 values to not up-case the `cdist` results in the same way that was happening in the `weight_regions_by_scalar` mask inversion.

Reply:
It looks like the `cdist` function automatically upcasts returned results (see example below).

```
from scipy.spatial.distance import cdist
coords = np.array([(35.0456, -85.2672),
          (35.1174, -89.9711),
          (35.9728, -83.9422),
          (36.1667, -86.7833)]).astype('float32')
print(cdist(coords, coords, 'euclidean').dtype)  # should be 'float64'
```

Should we just convert the `cdist` results to `float32`?

Reply:
If `cdist` is doing unknown casting and we want to make sure that out math stays within known type constraints, then yes. However, instead of just always casting to a constant, we should use `np.promote_types(T1, T2)` to determine the cast type at runtime, e.g.
```ipython
In [5]: np.promote_types(np.uint8, np.uint16)
Out[5]: dtype('uint16')
```